### PR TITLE
Reimplement "Addition of Testgrid Service Account"

### PR DESCRIPTION
**Pull request by chases2**
Reimplements #14364 (also see #14366)
Includes fixes to config-upload.sh to reference the correct file

/cc @michelle192837 @cjwagner @Katharine

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 1]
```diff
@@ -664,6 +664,40 @@ postsubmits:
...
           secretName: deployer-service-account
+  - name: post-test-infra-upload-testgrid-config
+    cluster: test-infra-trusted
+    branches:
+    - master
+    labels:
+      preset-bazel-scratch-dir: "true"
+    env:
+    - name: GOOGLE_APPLICATION_CREDENTIALS
+      value: /etc/service-account/service-account.json
```
Comment (cjwagner) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: I don't think the env is required since you are using a flag now.

**Comment by Katharine** [comment]
This isn't really a revert; please change the title accordingly.

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml (hunk 1 above)
Comment (chases2) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: /hold
I'm pretty sure you're right, but I want to make sure

Comment (chases2) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: Writing to TestGrid canary was successful using `pj-on-kind.sh`

```
INFO: Running command line: bazel-bin/testgrid/cmd/configurator/linux_amd64_pure_stripped/configurator '--gcp-service-account=/etc/service-account/service-account.json' '--yaml=/home/prow/go/src/github.com/chases2/test-infra/config/testgrids' '--default=/home/prow/go/src/github.com/chases2/test-infra/config/testgrids/default.yaml' '--prow-config=/home/prow/go/src/github.com/chases2/test-infra/prow/config.yaml' '--prow-job-config=/home/prow/go/src/github.com/chases2/test-infra/config/jobs/' '--output=gs://k8s-testgrid-canary/config' --oneshot --world-readable
```

/hold cancel

**Comment by chases2** [comment]
/kind oncall-hotlist
/hold cancel

Merge today, bump tomorrow

**Pull request merged**